hooks/useInitXmtpClient.ts

```diff
@@ -1,28 +1,52 @@
 import { Client } from '@xmtp/xmtp-js'
 import { Signer } from 'ethers'
 import { useCallback, useEffect, useState } from 'react'
-import { getEnv } from '../helpers'
+import {
+  getAppVersion,
+  getEnv,
+  loadKeys,
+  storeKeys,
+  wipeKeys,
+} from '../helpers'
 import { useAppStore } from '../store/app'
 
-const useInitXmtpClient = () => {
+const useInitXmtpClient = (cacheOnly = false) => {
   const signer = useAppStore((state) => state.signer)
+  const address = useAppStore((state) => state.address) ?? ''
   const client = useAppStore((state) => state.client)
   const setClient = useAppStore((state) => state.setClient)
   const reset = useAppStore((state) => state.reset)
   const [isRequestPending, setIsRequestPending] = useState(false)
 
-  const disconnect = () => {
+  const disconnect = async () => {
     setClient(undefined)
     reset()
+    if (signer) {
+      wipeKeys(address)
+    }
   }
 
   const initClient = useCallback(
     async (wallet: Signer) => {
       if (wallet && !client) {
         try {
-          setIsRequestPending(true)
-          const response = await Client.create(wallet, { env: getEnv() })
-          setClient(response)
+          let keys = loadKeys(address)
+          if (!keys) {
+            if (cacheOnly) {
+              return
+            }
+            keys = await Client.getKeys(wallet, {
+              env: getEnv(),
+              appVersion: getAppVersion(),
+            })
+            storeKeys(address, keys)
+          } else setIsRequestPending(true)

```

Comment:
Surprised our linter allows for an else statement without braces. 

We should probably add a rule against that.